rconweb/api/views.py

```diff
@@ -82,30 +82,44 @@ def get_version(request):
 
 @csrf_exempt
 def public_info(request):
-    status = ctl.get_status()
+    gamestate = ctl.get_gamestate()
+    curr_players, max_players = tuple(map(int, ctl.get_slots().split("/")))
     try:
-        current_map = MapsHistory()[0]
+        current_map_start = MapsHistory(max_len=1)[0]["start"]
     except IndexError:
         logger.error("Can't get current map time, map_recorder is probably offline")
-        current_map = {"name": status["map"], "start": None, "end": None}
-    current_map = dict(
-        just_name=map_name(current_map["name"]),
-        human_name=LONG_HUMAN_MAP_NAMES.get(current_map["name"], current_map["name"]),
-        **current_map,
-    )
-    vote_status = get_votes_status(none_on_fail=True)
-    next_map = ctl.get_next_map()
+        current_map_start = None
+
+    def explode_map_info(game_map: str, start) -> dict:
+        return dict(
+            just_name = map_name(game_map),
+            human_name = LONG_HUMAN_MAP_NAMES.get(game_map, game_map),
+            name = game_map,
+            start = start
+        )
+
     return api_response(
-        result=dict(
-            current_map=current_map,
-            **status,
-            vote_status=vote_status,
-            next_map=next_map,
+        result = dict(
```

Comment:
Is this backward compatible? I think it's used in the public gamestate view and in the scorebot.

Reply:
Good catch! Should have checked that beforehand but thought all the other things would use the "internal structure".
That's a dependency I did not think of, that does look somewhat cumbersome.
But I (hopefully) fixed those dependencies, which should make it "backward compatible" again.

Reply:
I'm not sure if it is or not. If it's just extra keys it probably is fine

Reply:
No the 'signature' of the dict had to change a bit. For example the inclusion of the `human_name` for the `next_map` (to have a format analog to the `current_map` format) does mean the 'internal' name of `next_map` is not anymore reachable via `next_map` but instead via `next_map.name`. But I did search for all occurrences of `public_info` and adapted the code accordingly.